ahn/training/_decoder.py

```diff
@@ -49,34 +51,140 @@ def tokenize(config, tokenizer, data, data_type):
             imap = tqdm(imap, total=len(data))
         for token in imap:
             outputs.append(token)
-
     return outputs
 
 
+def get_data_loader_hfstyle(config, tokenizer, split):
+    accelerator = Accelerator()
+    if split == "train":
+        dataset = datasets.load_dataset(
+            "json", data_files=config["data"]["train_data_path"], split="all"
+        )
+    elif split == "valid":
+        dataset = datasets.load_dataset(
+            "json", data_files=config["data"]["valid_data_path"], split="all"
+        )
+    else:
+        raise NotImplementedError(f"split {split} is not implemented")
+    if config.get("debug"):
+        dataset = dataset.select(range(100))
+
+    def process_fn(examples):
+        result = tokenizer(
+            examples[config["data"]["train_data_key"]],
+            padding=False,
+            truncation=False,
+            max_length=config["model_and_tokenizer"]["max_length"],
+            return_attention_mask=False,
+            return_token_type_ids=False,
+        )
+        return result
+
+    with accelerator.main_process_first():
+        dataset = dataset.map(
```

Comment:
what's the advantage of this?

Reply:
first of all, we don't need to retokenize because it uses caching after map. Also by using batch tokenizing, it would make faster.